lib/tests/test_revset.rs

```diff
@@ -1421,6 +1435,33 @@ fn test_evaluate_expression_remote_branches(use_git: bool) {
         resolve_commit_ids(mut_repo.as_repo_ref(), "remote_branches()"),
         vec![commit2.id().clone(), commit1.id().clone()]
     );
+    // Can get branches with matching names
+    assert_eq!(
+        resolve_commit_ids(mut_repo.as_repo_ref(), "remote_branches(branch1)"),
+        vec![commit1.id().clone()]
+    );
+    assert_eq!(
+        resolve_commit_ids(mut_repo.as_repo_ref(), "remote_branches(branch)"),
+        vec![commit2.id().clone(), commit1.id().clone()]
+    );
+    // Can get branches from matching remotes
+    assert_eq!(
+        resolve_commit_ids(mut_repo.as_repo_ref(), "remote_branches(\"\", origin)"),
+        vec![commit1.id().clone()]
+    );
+    assert_eq!(
+        resolve_commit_ids(mut_repo.as_repo_ref(), "remote_branches(\"\", ri)"),
+        vec![commit2.id().clone(), commit1.id().clone()]
+    );
+    // Can silently resolve to an empty set if there's no matches
+    assert_eq!(
+        resolve_commit_ids(mut_repo.as_repo_ref(), "remote_branches(branch3)"),
+        vec![]
+    );
+    assert_eq!(
+        resolve_commit_ids(mut_repo.as_repo_ref(), "remote_branches(\"\", upstream)"),
+        vec![]
+    );
```

Comment:
Can you add a test for `remote_branches(non_empty, non_empty)` case?

Reply:
Done.